test/distributed/org/apache/cassandra/distributed/test/accord/AccordCQLTest.java

```diff
@@ -266,6 +269,111 @@ private void testScalarIsNull(String tableDDL) throws Exception {
              });
     }
 
+    @Test
+    public void testQueryStaticColumn() throws Exception
+    {
+        test("CREATE TABLE " + currentTable + " (k int, c int, s int static, v int, primary key (k, c))",
+             cluster ->
+             {
+                 // select partition key, clustering key and static column, restrict on partition and clustering
+                 testQueryStaticColumn(cluster,
+                                       "LET row0 = (SELECT k, c, s, v FROM " + currentTable + " WHERE k = ? AND c = 0);\n" +
+                                       "SELECT row0.k, row0.c, row0.s, row0.v;\n",
+
+                                       "SELECT k, c, s, v FROM " + currentTable + " WHERE k = ? AND c = 0");
+
+                 // select partition key, clustering key and static column, restrict on partition and limit to 1 row
+                 testQueryStaticColumn(cluster,
+                                       "LET row0 = (SELECT k, c, s, v FROM " + currentTable + " WHERE k = ? LIMIT 1);\n" +
+                                       "SELECT row0.k, row0.c, row0.s, row0.v;\n",
+
+                                       "SELECT k, c, s, v FROM " + currentTable + " WHERE k = ? LIMIT 1");
+
+                 // select just static column, restrict on partition and clustering
```

Comment:
This selects s and v, not just s?

Reply:
```org.apache.cassandra.exceptions.InvalidRequestException: Cannot restrict clustering columns when selecting only static columns```

Reply:
Ah, yeah, in that case, makes sense, although I'd just update the comment?